Comments

ShyLee873 opened the pull request.

ShyLee873 committed:
working on adding comments to posts

ShyLee873 committed:
Created comments model with working create/destroy methods to be utilized by both posts and events through polymorphic associatit models through polymorphic association

ShyLee873 committed:
Created comments model to be used by post and event models through polymorphic association

ShyLee873 committed:
fixed errors and added comments model to use with both post and event models

ShyLee873 committed:
Merge branch 'master' into comments

kostik700015 commented on app/models/comment.rb:
```diff
@@ -0,0 +1,2 @@
+class Comment < ApplicationRecord
+end
```
Why did you remove these lines? I think we need them for the polymorphic association. 
```
`belongs_to :commentable, polymorphic: true`
`validates :body, presence: true`
```
For the polymorphic association you should add them to make association works.

Also we may need to add `has_many :comments` to User Model so User can have many comments

ShyLee873 replied:
The polymorphic association  is so that the Comment model can be integrated into both the post and events as discussed. I'm not sure what happened to the comment model, I had to pull stuff from the repo before I could make my changes. I'll fix it.

kostik700015 replied:
> The polymorphic association is so that the Comment model can be integrated into both the post and events as discussed. 

- I totally agree with it.  I was confused when noticed empty comment model.  Sure, after pulling last changes you will have opportunity to do that as we discussed with polymorphic model.  Let me know if you have any questions.

kostik700015 reviewed (changes requested):
I think that we actually don't need a polymorphic association between Post and Comment models. The polymorphic association is used when the `Model-A` can `belongs_to` either `Model-B` or `Model-C`. In our case Comments can belong both User and Post.
@jacobgwilliams , @Xxyumi-hub  what do you think?

kostik700015 commented:
I also got an error when ran `rails db:migrate`. I think it happened because it tried to delete non-existing yet table 'comments'(in order before '**comments**' table was created). After I deleted the migration file `20220201193625_drop_comments.rb` I could run migration and '**comments**' was created

ShyLee873 commented:
I'll have to push the changes after this is merged. Before, my local branch was behind for some reason and I had to take some corrective measures to fix the issue. the belongs_to :commentable, polymorphic: true line was overlooked in the repair. I'll have to add it back with the next push to avoid repeating the same mistakes.

ShyLee873 commented:
> I also got an error when ran `rails db:migrate`. I think it happened because it tried to delete non-existing yet table 'comments'(in order before '**comments**' table was created). After I deleted the migration file `20220201193625_drop_comments.rb` I could run migration and '**comments**' was created

Yep. this is all corrected with this latest PR. Again, my branch was behind and things were changed and deleted on my end. With this PR, my new comments table is functional.

ShyLee873 commented:
I'll just close and do it all again after I get my kids to bed. No worries.
